app/Http/Controllers/OAuthController.php

```diff
@@ -2,20 +2,67 @@
 
 namespace App\Http\Controllers;
 
-use App\Models\User;
-use CDash\Middleware\OAuth2\OAuth2Interface;
 use Illuminate\Http\RedirectResponse;
 use Illuminate\Http\Request;
 use Illuminate\Support\Collection;
 use Illuminate\Support\Facades\Auth;
+use Illuminate\Support\Facades\Hash;
+use Illuminate\Support\Facades\Log;
+use Illuminate\Support\Str;
+use Laravel\Socialite\Facades\Socialite;
+use Illuminate\Routing\Redirector;
+use Symfony\Component\HttpFoundation\RedirectResponse as symfonyResponse;
 use RuntimeException;
 
+use App\Models\User;
+use App\Http\Controllers\Auth\RegisterController;
+use CDash\Middleware\OAuth2\OAuth2Interface;
+
 /**
  * Class OAuthController
  * @package App\Http\Controllers
  */
 final class OAuthController extends AbstractController
 {
+    public function socialite(string $service): symfonyResponse
+    {
+        return Socialite::driver($service)->redirect();
+    }
+
+    public function callback(string $service): RedirectResponse|Redirector
+    {
+        $authUser = Socialite::driver($service)->user();
+        $email =  $authUser->getEmail();
+        [$fname, $lname] = explode(" ", $authUser->getName() ?? '');
+
+        // TODO: What if, for whatever reason, there is more than one user found?
+        $user = \App\Models\User::firstWhere('email', $email);
```

Comment:
You already included `User` via the `use` statement at the top of the file, so I think you can drop the full namespace here:

```suggestion
        $user = User::firstWhere('email', $email);
```

Reply:
Thanks! Fixed and updated.